desktop/NodeManager.ts

```diff
@@ -112,45 +112,53 @@ class NodeManager {
     const nodeDataFilesPath = path.resolve(`${userDataPath}`, 'node-data');
     const logFilePath = path.resolve(`${userDataPath}`, 'spacemesh-log.txt');
 
-    if (this.cleanStart) {
-      if (fs.existsSync(logFilePath)) {
-        fs.unlinkSync(logFilePath);
-      }
-      const nodePathWithParams = `"${nodePath}" --config "${this.configFilePath}" -d "${nodeDataFilesPath}" > "${logFilePath}"`;
-      this.nodeProcess = exec(nodePathWithParams, (error, _, stderr) => {
-        if (error) {
-          // Take the first line of stderr or fallback to error.message
-          const message = stderr.split('\n')[0] || error.message;
-          this.sendNodeError(normalizeCrashError(error, message));
-          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-          (process.env.NODE_ENV !== 'production' || process.env.DEBUG_PROD === 'true') && dialog.showErrorBox('Smesher Start Error', `${error}`);
-          logger.error('startNode', error);
-        }
-      });
-    } else {
-      const netId = StoreService.get('netSettings.netId');
-      const savedSmeshingParams = StoreService.get(`${netId}-smeshingParams`);
-      const nodePathWithParams = `"${nodePath}" --config "${this.configFilePath}"${
-        savedSmeshingParams ? ` --coinbase 0x${savedSmeshingParams.coinbase} --start-mining --post-datadir "${savedSmeshingParams.dataDir}"` : ''
-      } -d "${nodeDataFilesPath}" >> "${logFilePath}"`;
-      this.nodeProcess = exec(nodePathWithParams, (error, _, stderr) => {
-        if (error) {
-          // Take the first line of stderr or fallback to error.message
-          const message = stderr.split('\n')[0] || error.message;
-          this.sendNodeError(normalizeCrashError(error, message));
-          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-          (process.env.NODE_ENV !== 'production' || process.env.DEBUG_PROD === 'true') && dialog.showErrorBox('Smesher Error', `${error}`);
-          logger.error('startNode', error);
-        }
-      });
-    }
+    const logFileStream = fs.createWriteStream(logFilePath, { flags: this.cleanStart ? 'w' : 'a', encoding: 'utf-8' });
+    const netId = StoreService.get('netSettings.netId');
+    const savedSmeshingParams = StoreService.get(`${netId}-smeshingParams`);
+    const smeshingArgs = savedSmeshingParams ? ['--coinbase', `0x${savedSmeshingParams.coinbase}`, '--start-mining', '--post-datadir', savedSmeshingParams.dataDir] : [];
+    const args = ['--config', this.configFilePath, '-d', nodeDataFilesPath, ...smeshingArgs];
+
+    this.nodeProcess = spawn(nodePath, args);
```

Comment:
Can you please add logging to app logger with args? this will greatly ease troubleshooting users node startup issues.

Reply:
Done!